test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -41,31 +42,87 @@ public void NonGeneric_GetRequiredServices_ReturnsAllServices()
         }
 
         [Fact]
-        public void GetServicesReturns_A_Service()
+        public void GetRequiredServices_Returns_SingleService()
         {
             // Arrange
-            var serviceProvider = CreateFooServiceProvider(1);
+            var serviceProvider = CreateTestServiceProvider(1);
 
             // Act
             var services = serviceProvider.GetRequiredServices<IFoo>();
 
             // Assert
-            Assert.Contains(services, item => item is Foo1);
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
             Assert.Equal(1, services.Count());
         }
 
         [Fact]
-        public void ThrowsWhenNoServicesAreRegistered()
+        public void NonGeneric_GetRequiredServices_Returns_SingleService()
         {
             // Arrange
-            var serviceProvider = CreateFooServiceProvider(0);
+            var serviceProvider = CreateTestServiceProvider(1);
+
+            // Act
+            var services = serviceProvider.GetRequiredServices(typeof(IFoo));
+
+            // Assert
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
+            Assert.Equal(1, services.Count());
```

Comment:
You shouldn't need this - `Assert.Single` throws if the sequence has more than one element in it.